From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ThomasKaiser <github@kaiser-edv.de>
Date: Sat, 9 Apr 2016 16:02:53 +0200
Subject: [ARCHEOLOGY] Define safe voltage/cpufreq defaults for sun8i/dev

> X-Git-Archeology: - Revision da11a8c01fae54c17503d2ded7c0f972fd5d384a: https://github.com/armbian/build/commit/da11a8c01fae54c17503d2ded7c0f972fd5d384a
> X-Git-Archeology:   Date: Sat, 09 Apr 2016 16:02:53 +0200
> X-Git-Archeology:   From: ThomasKaiser <github@kaiser-edv.de>
> X-Git-Archeology:   Subject: Define safe voltage/cpufreq defaults for sun8i/dev
> X-Git-Archeology:
> X-Git-Archeology: - Revision dddf22832564dda0102ea07bcc56fbbefae3b050: https://github.com/armbian/build/commit/dddf22832564dda0102ea07bcc56fbbefae3b050
> X-Git-Archeology:   Date: Tue, 29 Nov 2016 14:27:57 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Merge and rename u-boot patch directories
> X-Git-Archeology:
> X-Git-Archeology: - Revision a6140729064068b611a546d0580cfc34ee561b8b: https://github.com/armbian/build/commit/a6140729064068b611a546d0580cfc34ee561b8b
> X-Git-Archeology:   Date: Sun, 11 Dec 2016 17:40:27 +0300
> X-Git-Archeology:   From: zador-blood-stained <zador-blood-stained@users.noreply.github.com>
> X-Git-Archeology:   Subject: Rename and remove obsolete u-boot patches
> X-Git-Archeology:
> X-Git-Archeology: - Revision 3cb7175c667da480e59ef5b9bb23e238b4955cdf: https://github.com/armbian/build/commit/3cb7175c667da480e59ef5b9bb23e238b4955cdf
> X-Git-Archeology:   Date: Thu, 15 Sep 2022 11:08:20 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Upgrade Allwinner boot loader to 2022.08 (#4168)
> X-Git-Archeology:
> X-Git-Archeology: - Revision f6a09def35e647a5442e0f92f399485be29f19a9: https://github.com/armbian/build/commit/f6a09def35e647a5442e0f92f399485be29f19a9
> X-Git-Archeology:   Date: Thu, 10 Nov 2022 21:49:36 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Moving patches to per board and removing obsolete (#4409)
> X-Git-Archeology:
---
 arch/arm/include/asm/arch-sunxi/clock_sun6i.h | 1 +
 arch/arm/mach-sunxi/clock_sun6i.c             | 8 +++++---
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/include/asm/arch-sunxi/clock_sun6i.h b/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
index 7fcf340db6..723113b10b 100644
--- a/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
+++ b/arch/arm/include/asm/arch-sunxi/clock_sun6i.h
@@ -208,6 +208,7 @@ struct sunxi_ccm_reg {
 #define CCM_PLL1_CTRL_N(n)		((((n) - 1) & 0x1f) << 8)
 #define CCM_PLL1_CTRL_P(n)		(((n) & 0x3) << 16)
 #define CCM_PLL1_CTRL_EN		(0x1 << 31)
+#define CCM_PLL1_CTRL_LOCK		(0x1 << 28)
 
 #define CCM_PLL3_CTRL_M_SHIFT		0
 #define CCM_PLL3_CTRL_M_MASK		(0xf << CCM_PLL3_CTRL_M_SHIFT)
diff --git a/arch/arm/mach-sunxi/clock_sun6i.c b/arch/arm/mach-sunxi/clock_sun6i.c
index 86f08ab717..0c403204c5 100644
--- a/arch/arm/mach-sunxi/clock_sun6i.c
+++ b/arch/arm/mach-sunxi/clock_sun6i.c
@@ -158,15 +158,17 @@ void clock_set_pll1(unsigned int clk)
 	writel(CCM_PLL1_CTRL_EN | CCM_PLL1_CTRL_P(p) |
 	       CCM_PLL1_CTRL_N(clk / (24000000 * k / m)) |
 	       CCM_PLL1_CTRL_K(k) | CCM_PLL1_CTRL_M(m), &ccm->pll1_cfg);
-	sdelay(200);
+
+	while (!(readl(&ccm->pll1_cfg) & CCM_PLL1_CTRL_LOCK))
+		;
 
 	/* Switch CPU to PLL1 */
 	if (IS_ENABLED(CONFIG_MACH_SUNIV)) {
 		writel(CPU_CLK_SRC_PLL1 << CPU_CLK_SRC_SHIFT,
 		       &ccm->cpu_axi_cfg);
 	} else {
-		writel(AXI_DIV_3 << AXI_DIV_SHIFT |
-		       ATB_DIV_2 << ATB_DIV_SHIFT |
+		writel(AXI_DIV_4 << AXI_DIV_SHIFT |
+		       ATB_DIV_4 << ATB_DIV_SHIFT |
 		       CPU_CLK_SRC_PLL1 << CPU_CLK_SRC_SHIFT,
 		       &ccm->cpu_axi_cfg);
 	}
-- 
Armbian

